pkg/nsx/services/inventory/inventory.go

```diff
@@ -0,0 +1,243 @@
+
+	InventoryStausUp       = "UP"
+	InventoryStatusDown    = "DOWN"
+	InventoryStatusUnknown = "UNKNOWN"
+)
+
+var (
+	log         = &logger.Log
+	emptyKeySet = sets.New[InventoryKey]()
+)
+
+type InventoryService struct {
+	commonservice.Service
+	applicationInstanceStore *ApplicationInstanceStore
+	clusterStore             *ClusteStore
+
+	requestBuffer []containerinventory.ContainerInventoryObject
+	pendingAdd    map[string]interface{}
+	pendingDelete map[string]interface{}
+
+	stalePods map[string]interface{}
+}
+
+func InitializeService(service commonservice.Service) (*InventoryService, error) {
+	inventoryService := &InventoryService{
+		requestBuffer: make([]containerinventory.ContainerInventoryObject, 0),
+		pendingAdd:    make(map[string]interface{}),
+		pendingDelete: make(map[string]interface{}),
+		stalePods:     make(map[string]interface{}),
+	}
+
+	// TODO, Inventory store should have its own store
+	inventoryService.applicationInstanceStore = &ApplicationInstanceStore{ResourceStore: commonservice.ResourceStore{
+		Indexer: cache.NewIndexer(keyFunc, cache.Indexers{string(ContainerApplicationInstance): indexFunc}),
+	}}
+	inventoryService.clusterStore = &ClusteStore{ResourceStore: commonservice.ResourceStore{
+		Indexer: cache.NewIndexer(keyFunc, cache.Indexers{string(ContainerCluster): indexFunc}),
+	}}
+	inventoryService.Service = service
+	return inventoryService, nil
+}
+
+func (s *InventoryService) Init() error {
+	cluster, err := s.GetContainerCluster()
+	// If there is no such cluster, create one.
+	// Otherwise, sync with NSX for different types of inventory objects.
+	if err == nil {
+		err = s.clusterStore.Add(&cluster)
+		if err != nil {
+			log.Error(err, "Add cluster to store")
+		}
+		return err
+	}
+	log.Error(err, "Cannot find existing container cluster, will create one")
+	newContainerCluster := s.BuildInventoryCluster()
+	cluster, err = s.AddContainerCluster(newContainerCluster)
+	if err != nil {
+		return err
+	}
+	err = s.clusterStore.Add(&cluster)
+	if err != nil {
+		log.Error(err, "Add cluster to store")
+		return err
+	}
+	log.Info("A new ContainerCluster is added", "cluster", newContainerCluster.DisplayName)
+	err = s.initContainerApplicationInstance(s.NSXConfig.Cluster)
+	if err != nil {
+		log.Error(err, "Populating inventory object from nsx error")
+	}
+
+	return nil
+}
+
+func (s *InventoryService) SyncInventoryStoreByType(clusterId string) error {
+	log.Info("Populating inventory object from NSX")
+	err := s.initContainerApplicationInstance(clusterId)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *InventoryService) SyncInventoryObject(bufferedKeys sets.Set[InventoryKey]) (sets.Set[InventoryKey], error) {
+	retryKeys := sets.New[InventoryKey]()
+	startTime := time.Now()
+	defer func() {
+		log.Info("Finished syncing inventory object", "duration", time.Since(startTime))
+	}()
+	for key := range bufferedKeys {
+		log.V(1).Info("Syncing inventory object", "object key", key)
```

Comment:
log.V(1).Info -> log.Info, since it is important.

Reply:
Done